### PR TITLE
Added dark and light to Drop and Layer in theme

### LINKED ISSUE #4160: DropButton Content Doesn't Respond to Dark Mode
### Expected Behavior

When a DropButton element is clicked while the app's theme is set to "dark" in the defaultMode property of the theme, the background of the Box should be dark.

### Actual Behavior

When a DropButton is clicked while the app is in dark mode, the dropContent's background remains light.

### URL, screen shot, or Codepen exhibiting the issue

CodePen URl:
https://codesandbox.io/s/grommet-v2-template-xe041

Click on the Example DropButton, and you'll see that the dropContent has a light background, despite the app being in dark mode.

### Steps to Reproduce

1. Create a simple Grommet app.
2. Set a theme to use `defaultMode: "dark"`, then make the top-level Grommet container use that theme, as opposed to the default `grommet` theme.
3. Click on the DropButton, and you'll see that the dropContent is light.

### Your Environment

As in the CodeSandbox, these are the dependency versions:
- grommet: 2.12.0
- grommet-icons: 4.4.0
- react: 16.6.1
- react-dom: 16.6.1
- styled-components: 5.0.1

## Side Note
Grommet is seriously awesome. Thank you so much guys for creating an incredible replacement to Material Design!

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Previously the background for Drop and Layer was always set to white. This PR changes the theme so that there is dark and light for the Drop and Layer background.

#### Where should the reviewer start?
themes/base.js

#### What testing has been done on this PR?
Tested with the following stories:
```javascript
import React from 'react';
import { storiesOf } from '@storybook/react';

import { Box, Grommet, Layer, Text } from 'grommet';
import { grommet } from 'grommet/themes';

const customTheme = {
    defaultMode: "dark",
    global: { ...grommet.global }
  };

const NotificationLayer = () => {
  return (
    <Grommet theme={customTheme} full>
      <Layer margin="medium">
        <Box pad="large" border={{ color: 'accent-1', size: 'large' }}>
          <Text color="accent-2">Text</Text>
        </Box>
      </Layer>
    </Grommet>
  );
};

storiesOf('Layer', module).add('TEST', () => <NotificationLayer />);
```

```javascript
import { storiesOf } from '@storybook/react';
import React from "react";
import { grommet, Box, Grommet, Paragraph, DropButton } from "grommet";

const customTheme = {
    defaultMode: "dark",
    global: { ...grommet.global }
  };

const SimpleDropButton = () => {
  return (
    <Grommet theme={customTheme}>
        <Box pad="small">
          <DropButton
            label="Example DropButton"
            dropAlign={{ top: "bottom", right: "right" }}
            dropContent={<Box pad="large"/>}
          />
        </Box>
        <Paragraph>
          As seen in the above DropButton, the dropContent does not adapt to
          dark mode. The background remains light. The code is from the
          documentation, https://v2.grommet.io/dropbutton. Click the DropButton
          and you'll see that the background remains light.
        </Paragraph>
      </Grommet>
  );
};

storiesOf('DropButton', module).add('TEST', () => <SimpleDropButton />, {
  chromatic: { disable: true },
});
```

#### What are the relevant issues?
Closes #4160 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
Maybe

#### Is this change backwards compatible or is it a breaking change?
This may be a breaking change, I am a little worried about the snapshot changes and possible changes in the stories in storybook

### CODE REVIEW

File: src/js/components/Layer/__tests__/__snapshots__/Layer-test.js.snap [hunk 1]
```diff
@@ -900,8 +900,8 @@ exports[`Layer dark context 1`] = `
   -ms-flex-direction: column;
   flex-direction: column;
   min-height: 48px;
-  background-color: #FFFFFF;
-  color: #444444;
+  background-color: #000000;
+  color: #f8f8f8;
```
Comment (ShimiSun) [inline src/js/components/Layer/__tests__/__snapshots__/Layer-test.js.snap:904]: This snapshot change is suspicious, can you please verify it's the correct behavior for the test?

Comment (jcfilben) [reply src/js/components/Layer/__tests__/__snapshots__/Layer-test.js.snap:904]: I believe that this is the correct behavior. Previously in the test the Box has a dark background and the Layer has a light background with dark text. Now the Box and Layer have a dark background and the text in the Layer is light.

Comment (jcfilben) [reply src/js/components/Layer/__tests__/__snapshots__/Layer-test.js.snap:904]: Screen Shot 2020-10-26 at 11 51 02 AM

**Comment by ericsoderberghp** [comment]
I am thinking this looks OK. But, I wanted to verify that if the caller calls `<Grommet />` and sets the `themeMode` prop, that the Drop and Layer backgrounds will respect that. Is that so?

**Comment by jcfilben** [comment]
Layer when `themeMode` is switched to dark:

LayerDark

Drop when `themeMode` is switched to dark:

DarkDrop

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**